app/actions/websocket/users.test.ts

```diff
@@ -0,0 +1,280 @@
+
+            jest.mocked(fetchUsersByIds).mockResolvedValue({
+                users: [],
+                existingUsers: [],
+            });
+            const msg = {
+                data: {
+                    user_id: otherUserId,
+                },
+                broadcast: {
+                    channel_id: 'channel-id',
+                },
+            } as WebSocketMessage;
+            jest.useFakeTimers();
+            const promise = handleUserTypingEvent(serverUrl, msg);
+            jest.runAllTimers();
+            jest.useRealTimers();
+            await promise;
+            expect(DeviceEventEmitter.emit).toHaveBeenCalledWith(Events.USER_TYPING, expect.any(Object));
+        });
+
+        it('should emit typing events', async () => {
+            jest.spyOn(DeviceEventEmitter, 'emit');
+            const channelId = 'channel-id';
+            const mockConfig = {
+                TimeBetweenUserTypingUpdatesMilliseconds: '500',
+            };
+
+            jest.mocked(getConfig).mockResolvedValue(mockConfig as any);
+            jest.mocked(fetchUsersByIds).mockResolvedValue({
+                users: [{id: otherUserId, username: 'other-user'} as UserProfile],
+                existingUsers: [],
+            });
+
+            jest.mocked(queryDisplayNamePreferences).mockReturnValue({
+                fetch: jest.fn().mockResolvedValue([{
+                    value: 'full_name',
+                }]),
+            } as any);
+
+            jest.mocked(getLicense).mockResolvedValue({} as ClientLicense);
+
+            const msg = {
+                data: {
+                    user_id: otherUserId,
+                    parent_id: 'root-id',
+                },
+                broadcast: {
+                    channel_id: channelId,
+                },
+            } as WebSocketMessage;
+
+            jest.useFakeTimers();
+            const promise = handleUserTypingEvent(serverUrl, msg);
+            jest.runAllTimers();
+            jest.useRealTimers();
+            await promise;
+
+            expect(DeviceEventEmitter.emit).toHaveBeenCalledWith(Events.USER_TYPING, expect.any(Object));
+
+            expect(DeviceEventEmitter.emit).toHaveBeenCalledWith(Events.USER_STOP_TYPING, expect.any(Object));
+        });
+    });
+
+    describe('userTyping', () => {
+        it('should send typing event', async () => {
+            const mockClient = {
+                sendUserTypingEvent: jest.fn(),
+            };
+            jest.mocked(WebsocketManager.getClient).mockReturnValue(mockClient as any);
+
+            await userTyping(serverUrl, 'channel-id', 'root-id');
+            expect(mockClient.sendUserTypingEvent).toHaveBeenCalledWith('channel-id', 'root-id');
+        });
+
+        it('should handle missing client', async () => {
+            jest.mocked(WebsocketManager.getClient).mockReturnValue(undefined);
+            await userTyping(serverUrl, 'channel-id');
+
+            // Should not throw
```

Comment:
we should remove this, unless we are handling them no test should throw

Reply:
We're testing the `client?.` part of the function for when client is undefined

Reply:
I don't know that comment feels weird to me, but I guess it is trying to explain the reason for the test to exist

Reply:
Yeah, I agree it's a little odd but I couldn't think of a better way to cover that branch